app/code/Magento/Customer/Test/Mftf/Test/StorefrontVerifyCustomCustomerAddressAttributeTest.xml

```diff
@@ -0,0 +1,144 @@
+        <selectOption stepKey="selectState" userInput="{{CustomerAddressSimple.state}}" selector="{{AdminEditCustomerAddressesSection.state}}"/>
+        <fillField  stepKey="fillCity" userInput="{{CustomerAddressSimple.city}}" selector="{{AdminEditCustomerAddressesSection.city}}"/>
+        <fillField  stepKey="fillZipCode" userInput="{{CustomerAddressSimple.postcode}}" selector="{{AdminEditCustomerAddressesSection.zipCode}}"/>
+        <fillField  stepKey="fillPhone" userInput="{{CustomerAddressSimple.telephone}}" selector="{{AdminEditCustomerAddressesSection.phone}}"/>
+        <actionGroup ref="SelectDropdownCustomerAddressAttributeValueActionGroup" stepKey="selectOptionValue1">
+            <argument name="customerAddressAttribute" value="AttributeDropdownData"/>
+            <argument name="optionValue" value="{{AttributeDropdownData.firstOption}}"/>
+        </actionGroup>
+        <waitForPageLoad stepKey="waitForAddressSaved"/>
+        <actionGroup ref="AdminLogoutActionGroup" stepKey="logout"/>
+
+        <!--Login To Store Front By B2C Customer-->
+        <actionGroup ref="LoginToStorefrontActionGroup" stepKey="loginAsStoreFront">
+            <argument name="Customer" value="$$simpleCustomer$$"/>
+        </actionGroup>
+        <!-- Add Product To Cart From Product Detail Page-->
+        <actionGroup ref="StorefrontOpenProductEntityPageActionGroup" stepKey="openProductPageOnStorefront">
+            <argument name="product" value="$$simpleProduct$$"/>
+        </actionGroup>
+        <actionGroup ref="AddToCartFromStorefrontProductPageActionGroup" stepKey="addToCartFromStorefrontProductPage">
+            <argument name="productName" value="$$simpleProduct.name$$"/>
+        </actionGroup>
+
+        <!--Go To Checkout and Verify Default Address-->
+        <actionGroup ref="GoToCheckoutFromMinicartActionGroup" stepKey="goToCheckoutFrom"/>
+        <!-- Ensure that the selected shipping address is similar to first address -->
+        <grabTextFrom selector="{{CheckoutShippingSection.selectedShippingAddress}}" stepKey="grabSelectedShippingAddress"/>
+        <assertStringContainsString stepKey="assertShippingAddressStreet">
+            <expectedResult type="string">{{CustomerAddressSimple.street[0]}}</expectedResult>
+            <actualResult type="variable">$grabSelectedShippingAddress</actualResult>
+        </assertStringContainsString>
+        <!--Verify that selected "Test Dropdown" options appears on the page-->
+        <assertStringContainsString stepKey="assertShippingAddressCustomAttribute">
+            <expectedResult type="string">{{AttributeDropdownData.firstOption}}</expectedResult>
+            <actualResult type="variable">$grabSelectedShippingAddress</actualResult>
+        </assertStringContainsString>
+
+        <!--Add Second Shipping Address-->
+        <click selector="{{CheckoutShippingSection.newAddressButton}}" stepKey="ClickOnAddNewAddressButton"/>
```

Comment:
change this & below waitForPageLoad to be an actiongroup instead
name: `StorefrontClickAddNewAddressButtonFromCheckoutShippingActionGroup`
description: `Clicks the New Address button on the storefront Checkout Shipping page`